superset-frontend/src/components/Datasource/DatasourceModal.tsx

```diff
@@ -140,7 +140,7 @@ const DatasourceModal: FunctionComponent<DatasourceModalProps> = ({
       cache_timeout:
         datasource.cache_timeout === '' ? null : datasource.cache_timeout,
       is_sqllab_view: datasource.is_sqllab_view,
-      template_params: datasource.template_params,
+      template_params: datasource.template_params === '' ? null : datasource.template_params,
```

Comment:
ideally we align the types and avoid the lazy `Record<string, any>` set above, seems the right type is here -> https://github.com/apache/superset/blob/master/superset-frontend/src/features/datasets/types.ts#L72 , let's use it at line 123

Now should template_params be a string? Unclear, maybe, maybe not, ideally the components where that came from always returns a proper JSON serializable, but that's probably more work. 

Now if the backend is in charge of serializing the string, if feels like the fix should be done there ideally. Not again the quickfix here but should include a `// TODO change type to {...} and push validation in the component`.

Trying to figure out where in the backend the serialization is done ...do you have a stacktrace of the backend failing?